header-only/tests/test_1494-layout-builder.cpp

```diff
@@ -531,7 +532,8 @@ test_ListOffset_Empty() {
           "\"class\": \"ListOffsetArray\", "
           "\"offsets\": \"i64\", "
           "\"content\": { "
-              "\"class\": \"EmptyArray\" "
+              "\"class\": \"EmptyArray\", "
+              "\"parameters\": {} "
```

Comment:
@ManasviGoyal why does this test change? Was it not being run before?

Reply:
When I run the the `LayoutBuilder` tests, the assertion is failing due to the missing `"parameters: {}"` described in the `EmptyArray` form. It's strange that it doesn't give any error while running the awkward tests.

Reply:
None of the Forms should explicitly require a `parameters` key when reading from JSON. The `parameters` key is always extracted with `get`:

https://github.com/scikit-hep/awkward/blob/089f8430e4b95b8b6b84ce941f6d1423dd6f4518/src/awkward/forms/form.py#L36-L42

(`ak.forms.from_json` immediately calls `ak.forms.from_dict`. When `parameters` are `None`, it's the same as if they're `{}`.)

On top of that, EmptyArray is required to not have any `parameters`. Explicitly providing `None` or `{}` is benign, though:

https://github.com/scikit-hep/awkward/blob/089f8430e4b95b8b6b84ce941f6d1423dd6f4518/src/awkward/forms/emptyform.py#L15-L21

If these are C++ tests that are failing, and the C++ tests are more restrictive than what the Python tests require, then it's the test that should change.

Reply:
Then the parameters part from the `EmptyArray` form in the `Layoutbuilder.h` code can be removed.

https://github.com/scikit-hep/awkward/blob/aba0b8d5412a1e0db89a27d065ab5f2bb6759846/header-only/layout-builder/awkward/LayoutBuilder.h#L416-L419

Reply:
Correct. It's also the only node type that sets it to a constant; all of the other LayoutBuilder node types take a `parameters` argument to pass on here.

Reply:
Yes exactly. I'll update the PR with the required changes.